gnomad_qc/v4/assessment/generate_ld_data.py

```diff
@@ -0,0 +1,566 @@
+    :param test: Filter to test versions and/or write to test paths.
+    """
+
+    for label, pops in dict(pop_data).items():
+        for pop in pops:
+            logger.info(f"Filtering for {pop}...")
+            pop_mt = filter_mt_for_ld(mt, label, pop, re_call_stats)
+            logger.info(f"Count after filter_mt_for_ld() : {pop_mt.count()}")
+
+            pruned_ht = hl.ld_prune(pop_mt.GT, r2=float(r2), bp_window_size=radius)
+
+            logger.info(f"Count after hl.ld_prune() : {pruned_ht.count()}")
+
+            ht = pop_mt.rows().select("pop_freq")
+            ht = ht.filter(hl.is_defined(pruned_ht[ht.key]))
+            ld_path = ld_pruned_path(data_type, pop, r2, version=version, test=test)
+            logger.info(f"Writing pruned ht for {pop} to {ld_path} ...")
+            ht.write(ld_path, overwrite)
+
+
+def generate_ld_matrix(
+    mt,
+    pop_data,
+    data_type,
+    radius: int = 1000000,
+    common_only: bool = True,
+    adj: bool = False,
+    overwrite: bool = False,
+    re_call_stats: bool = False,
+    version: str = None,
+    test: bool = False,
+) -> None:
+    """
+    Generate and write matrix of LD correlcations as Hail BlockMatrix using hl.ld_matrix. Read by generate_ld_scores_from_ld_matrix()
+
+    :param mt: Input MT to filter and generate LDs for
+    :param pop_data: Dict of population counts
+    :param data_type: Genetic data type for exomes or genomes. Exomes are currently too sparse, only runs on genomes.
+    :param radius: Used for bp_window_size, via Hail: Window size in base pairs (inclusive upper bound).
+    :param common_only: Bool of whether to filter to only common variants (>0.5%)
+    :param adj: Whether to use adj ("adjusted" or "passing") frequency
+    :param overwrite: Bool to write over previous outputs or not
+    :param re_call_stats: Bool to re-calculate callstats. Needed for subsets and when sampling cols.
+    :param version: Version of files, either 'hgdp' or None, which ld_pruned_path() populates with most recent gnomAD genomes version.
+    :param test: Filter to test versions and/or write to test paths.
+    """
+    # Takes about 4 hours on 20 n1-standard-8 nodes (with SSD - not sure if necessary) per population
+    # Total of ~37 hours ($400)
+
+    for label, pops in dict(pop_data).items():
+        for pop in pops:
+
+            pop_mt = filter_mt_for_ld(mt, label, pop, common_only, re_call_stats)
+
+            pop_mt.rows().select("pop_freq").add_index().write(
+                ld_index_path(
+                    data_type=data_type,
+                    pop=pop,
+                    common_only=common_only,
+                    adj=adj,
+                    version=version,
+                    test=test,
+                ),
+                overwrite,
+            )
+            ld = hl.ld_matrix(pop_mt.GT.n_alt_alleles(), pop_mt.locus, radius)
```

Comment:
In my old script, I ran `BlockMatrix.write_from_entry_expr(
            mt.GT.n_alt_alleles(), tmp_bm_path, mean_impute=True, center=False, normalize=False, overwrite=args.overwrite
        )`, wondering how much difference this will introduce